packages/jest-snapshot/src/State.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+'use strict';
+
+import type {Path} from 'types/Config';
+
+const {SnapshotFile, forFile} = require('./SnapshotFile');
+
+class SnapshotState {
+  _counters: Map<string, number>;
+  _index: number;
+  added: number;
+  matched: number;
+  snapshot: SnapshotFile;
+  unmatched: number;
+  update: boolean;
+  updated: number;
+
+  constructor(testPath: Path, update: boolean) {
+    this._counters = new Map();
+    this._index = 0;
+    this.added = 0;
+    this.matched = 0;
+    this.snapshot = forFile(testPath);
+    this.unmatched = 0;
+    this.update = update;
+    this.updated = 0;
+  }
+
+  match(testName: string, received: any) {
+    this._counters.set(testName, (this._counters.get(testName) || 0) + 1);
+    const count = Number(this._counters.get(testName));
+
+    const key = testName + ' ' + count;
```

Comment:
i see. we can probably dependency inject `key` here and do something like
```js
match(received: any, testName?: string, key?: string) {
  // ...
  key || (key = `${testName} ${count}`);
}
```

that would make it generic and reusable. would that work?